dotnet/src/SemanticKernel/CoreSkills/TextMemorySkill.cs

```diff
@@ -109,4 +147,30 @@ public async Task SaveAsync(string text, SKContext context)
 
         await context.Memory.SaveInformationAsync(collection, text: text, id: key);
     }
+
+    /// <summary>
+    /// Remove specific memory
+    /// </summary>
+    /// <example>
+    /// SKContext[TextMemorySkill.KeyParam] = "countryInfo1"
+    /// {{memory.remove }}
+    /// </example>
+    /// <param name="context">Contains the 'collection' containing the memory to remove.</param>
+    [SKFunction("Remove specific memory")]
+    [SKFunctionName("Remove")]
+    [SKFunctionContextParameter(Name = CollectionParam, Description = "Memories collection associated with the memory to remove",
+        DefaultValue = DefaultCollection)]
+    [SKFunctionContextParameter(Name = KeyParam, Description = "The key associated with the memory to remove")]
+    public async Task RemoveAsync(SKContext context)
+    {
+        var collection = context.Variables.ContainsKey(CollectionParam) ? context[CollectionParam] : DefaultCollection;
+        Verify.NotEmpty(collection, "Memory collection not defined");
+
+        var key = context.Variables.ContainsKey(KeyParam) ? context[KeyParam] : string.Empty;
+        Verify.NotEmpty(key, "Memory key not defined");
```

Comment:
another candidate for a helper, e.g. `GetRequired<T>(key)`

Reply:
see previous comment